msos/NativeMethods.cs

```diff
@@ -1,22 +1,171 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
 using System.Runtime.InteropServices;
-using System.Text;
-using System.Threading.Tasks;
+using Microsoft.Win32.SafeHandles;
+using FileTime = System.Runtime.InteropServices.ComTypes.FILETIME;
+using static msos.NativeStructs;
 
 namespace msos
 {
-    using FileTime = System.Runtime.InteropServices.ComTypes.FILETIME;
-
     static class NativeMethods
     {
+        #region DbgHelp
+        //public const int ERR_ELEMENT_NOT_FOUND = 1168;
+
+        [DllImport("Dbghelp.dll", SetLastError = true)]
+        public static extern bool MiniDumpReadDumpStream(IntPtr BaseOfDump,
+            MINIDUMP_STREAM_TYPE StreamNumber,
+            ref MINIDUMP_DIRECTORY Dir,
+            ref IntPtr StreamPointer,
+            ref uint StreamSize);
+
+        #endregion
+
+
+        #region Kernel32.dll
+
         [DllImport("kernel32.dll", SetLastError = true, CallingConvention = CallingConvention.Winapi)]
         [return: MarshalAs(UnmanagedType.Bool)]
         internal static extern bool IsWow64Process([In] IntPtr process, [Out] out bool wow64Process);
 
         [DllImport("kernel32.dll")]
         internal static extern bool GetProcessTimes(IntPtr hProcess, out FileTime lpCreationTime,
             out FileTime lpExitTime, out FileTime lpKernelTime, out FileTime lpUserTime);
+
+        [DllImport("kernel32.dll", SetLastError = true)]
+        public static extern IntPtr GetCurrentProcess();
+
+        [DllImport("kernel32.dll", SetLastError = true)]
+        public static extern IntPtr VirtualQuery(SafeMemoryMappedViewHandle address, ref MEMORY_BASIC_INFORMATION buffer, IntPtr sizeOfBuffer);
+
+
+        [DllImport("kernel32.dll", SetLastError = true)]
+        public static extern SafeMemoryMappedViewHandle MapViewOfFile(
```

Comment:
I think this also applies here, and to some other functions in this class as well -- so please review them all.